generate_markdown_from_doxygen_xml.py

```diff
@@ -0,0 +1,1031 @@
+        link_url=aTag.attrib['url']
+        #print('link_url: %s' % link_url)
+        link_text=lead_text+child_text
+        lead_text='[%s](%s)' % (link_text,link_url)
+        #print('lead_text: %s' % lead_text)
+        if html:
+            lead_text='<a href="%s">%s</a>' % (link_url,link_text)
+        tag_text=lead_text+tail_text #note, child text included in the URL text above. There won't be any though.
+        #print('tag_text: %s' % tag_text)
+        
+    else:
+        tag_text=lead_text+child_text+tail_text
+        
+    #print('debug:end:markdown_any_tag - tag_text: %s' % tag_text)
+    return tag_text
+        
+
+
+
+
+
+class cppAttribute:
+    def __init__(self):
+        self.kind='UNKNOWN'
+        self.name = 'UNKNOWN'
+        self.id = 'UNKNOWN'
+        self.prot= 'UNKNOWN'
+        self.static = 'UNKNOWN'
+        self.mutable = 'UNKNOWN'
+
+        self.type='UNKNOWN'
+        self.definition='UNKNOWN'
+        self.argsstring='UNKNOWN'
+        self.initializer=''
+        self.briefdescription='UNKNOWN'
+        self.detaileddescription='UNKNOWN'
+        self.inbodydescription='UNKNOWN'
+
+    def import_from_doxygen_class_file(self, aTag):
+        #print(aTag.attrib)
+        self.name = aTag.find('name').text
+        #print('attrib: self.name: %s' % self.name)
+        self.id = aTag.attrib['id']
+        self.prot= aTag.attrib['prot']
+        self.static = aTag.attrib['static']
+        self.kind = aTag.attrib['kind']
+        self.mutable = aTag.attrib['mutable']
+
+        self.type = aTag.find('type')
+        self.definition = aTag.find('definition')
+        self.argsstring = aTag.find('argsstring') 
+        self.initializer = aTag.find('initializer')
+        self.briefdescription = aTag.find('briefdescription')  # Needs for processing
+        self.detaileddescription = aTag.find('detaileddescription') # Needs for processing
+        self.inbodydescription = aTag.find('inbodydescription') # Needs for processing
+        
+    def markdown_overview(self):
+        #print('debug:start:cppAttribute:markdown_overview')
+        static_string=''
+        if self.static=='yes':
+            static_string='static'
+        output_string=''
+        #print('debug: static_string: %s' % static_string)
+        #print('debug: self.type: %s' % self.type)
+        #print('debug: self.name: %s' % self.name)
+        #print('debug: self.initializer: %s' % self.initializer)
+        #print('debug: self.briefdescription: %s' % markdown_any_tag(self.briefdescription).strip() )
+        
+        output_string+='\n\n%s %s [%s](#%s)%s - %s\n\n' % (static_string,markdown_any_tag(self.type).strip(), self.name,self.id,markdown_any_tag(self.initializer).strip(),markdown_any_tag(self.briefdescription).strip() )
+        return output_string
+        
+        
+    def markdown(self):
+        """
+        static_string=''
+        if self.static=='yes':
+            static_string='static '
```

Comment:
I don't understand this comment.

Reply:
It was commented out development code. Will delete.